querygen/querygen.go

```diff
@@ -57,3 +57,32 @@ func GenBuildPipelineRunStartedQuery(index string) string {
 	)
 	return q
 }
+
+// GenBuildTaskRunCompletedQuery returns a Splunk query for generating Segment events
+// when the clair-scan task completes.
+func GenBuildTaskRunCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="tekton.dev" `+
+			`"objectRef.resource"="taskruns" `+
+			`"objectRef.subresource"="status" `+
+			`"requestObject.metadata.labels.tekton.dev/pipelineTask"="clair-scan" `+
+			`"responseObject.status.completionTime"="*" `+
+			`| spath path=responseObject.status.conditions{} output=conditions `+
+			`| mvexpand conditions `+
+			`| spath input=conditions `+
+			`| where type="Succeeded" AND reason="Succeeded" AND status="True" `+
+			`| spath path=responseObject.status.taskResults{} output=taskResults `+
+			`| mvexpand taskResults `+
+			`| where name="CLAIR_SCAN_RESULT" `+
+			`| eval vulnerabilities_critical=mvindex('responseObject.status.taskResults{}.value.vulnerabilities.critical', 0),
+					vulnerabilities_high=mvindex('responseObject.status.taskResults{}.value.vulnerabilities.high', 0),
+					vulnerabilities_medium=mvindex('responseObject.status.taskResults{}.value.vulnerabilities.medium', 0),
+					vulnerabilities_low=mvindex('responseObject.status.taskResults{}.value.vulnerabilities.low', 0) `+
+			`| fields vulnerabilities_critical, vulnerabilities_high, vulnerabilities_medium, vulnerabilities_low `,
```

Comment:
We shouldn't declare fields like this. Instead, take a look at the `UserJourneyFieldSet` structure in `querygen/userjourney_fs.go` for how to do this part.

Reply:
@amisstea I followed `UserJourneyFieldSet` structure and created the respected fields. PLAT and let me know your suggestions.